agent/loop.py

```
@@ -1684,13 +1615,41 @@ def _resolve_free_nvidia_brain():
                 _wait = 2 ** _attempt  # 1s, 2s, 4s
                 log.warning("NVIDIA NIM rate-limited (HTTP %d) — retrying in %ds (attempt %d/%d)",
                            resp.status_code, _wait, _attempt + 1, _max_retries)
+                # Record 429 as a provider failure so the brain watchdog can
+                # trigger a failover if rate-limiting is sustained. The watchdog
+                # thresholds at 3 failures — a single 429 won't failover, but
+                # 3 consecutive 429s will switch to Ollama (local, no rate limit).
+                try:
+                    from packages.ai.watchdog import get_watchdog
+                    get_watchdog().record_failure("nvidia-nim")
+                except Exception:
+                    pass
```

Ruff (0.15.20)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)